### PR TITLE
Refactor/merge archive into core

### PR DESCRIPTION
**Pull request by hmaciasc**
This PR moves and merges archive into core simplifying our sysadmin tasks.
Before pushing to prod:
- Move DB file outside of archive and into core DB folder
- Renaming static_file folders into:
- [ ] staticData/archive_blobs/
- [ ] staticData/archive_thumbs/
- Reinstall core packages once merged

### CODE REVIEW

File: ipol_demo/modules/core/archive/archive.py [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+                conn.rollback()
+                conn.close()
+
+                # Execute deletion of copied files
+                for copied_file in copied_files_list:
+                    os.remove(copied_file)
+
+            except Exception:
+                pass
+
```
Comment (kidanger) [inline ipol_demo/modules/core/archive/archive.py:167]: Missing a return for the error case. Used here https://github.com/ipol-journal/ipolDevel/pull/196/files#diff-67df7a992db8fdc8199e09bd24943d112592476d57ef394361ea29df5057fc60R223

File: ipol_demo/modules/core/archive/archive.py [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+        if not os.path.isfile(settings.database_file):
+            try:
+                conn = lite.connect(settings.database_file)
+                cursor_db = conn.cursor()
+
+                sql_buffer = ""
+
+                with open(
+                    settings.database_dir + "/drop_create_db_schema.sql", "r"
```
Comment (kidanger) [inline ipol_demo/modules/core/archive/archive.py:529]:
```suggestion
               curdir = os.path.dirname(__file__)
                with open(
                    os.path.join(curdir, "drop_create_db_schema.sql"), "r"
```
(same as in demoinfo)
and remove the `database_dir` entry from the settings

File: ipol_demo/modules/core/config.py [hunk 3]
```diff
@@ -24,20 +24,22 @@ class Settings(BaseSettings):
...
 
     # dispatcher
     demorunners_path = (
         os.path.expanduser("~")
         + "/ipolDevel/ipol_demo/modules/config_common/demorunners.xml"
     )
     policy: str = "lowest_workload"
+    # archive
+    blobs_dir: str = "staticData/archive_blobs/"
+    blobs_thumbs_dir: str = "staticData/archive_thumbs/"
```
Comment (kidanger) [inline ipol_demo/modules/core/config.py:39]: can we prefix these with `archive_`? otherwise it will clash with the blobs module.

**Review by kidanger — CHANGES_REQUESTED** [review]

**Review by kidanger — APPROVED** [review]

**Review by mcolom — COMMENTED** [review]
No major comments on this PR, it looks good.
I have just two questions before accepting:
It seems that the static data from the Archive now will be in the Core's directory.
		- alias  /home/ipol/ipolDevel/ipol_demo/modules/archive/staticData/;
		+ alias  /home/ipol/ipolDevel/ipol_demo/modules/core/staticData/;
However, even if now you're importing Archive as a Core's component, still I think this static data should be in the Archive's directory for a better organization.

- You changed  BASE_URL = os.environ["IPOL_URL"] to BASE_URL = os.environ.get("IPOL_URL"). I'm curious to know why. Is it because one raises and exception and the other returns None, or something like that?

**Review by mcolom — APPROVED** [review]
OK, thanks for your answer!

**Review by Jyotsnarajan — APPROVED** [review]

**Pull request merged**